src/librustc/middle/infer/error_reporting.rs

```diff
@@ -486,13 +494,23 @@ impl<'a, 'tcx> ErrorReporting<'tcx> for InferCtxt<'a, 'tcx> {
             }
         };
 
+        let is_simple_error = if let &TypeError::Sorts(ref values) = terr {
```

Comment:
I don't understand what TypeError::Sorts for. Maybe other kinds also give the repetitive pattern, I will see.